### PR TITLE
Use omrstr_current_time_zone in javadump, emit time zone in the jitserver verbose log

### PR DESCRIPTION
**Pull request by cjjdespres**
The javadump time zone code is now redundant; the `omrstr_current_time_zone` function is used instead.

We now emit the time zone in the jitserver verbose log.

Blocked by: https://github.com/eclipse/omr/pull/6309
Fixes: https://github.com/eclipse-openj9/openj9/issues/13563

### CODE REVIEW

File: runtime/compiler/runtime/Listener.cpp [hunk 1]
```diff
@@ -348,9 +348,38 @@ static int32_t J9THREAD_PROC listenerThreadProc(void * entryarg)
...
+      char zoneName[32];
+      int32_t zoneSecondsEast = 0;
+      TR_VerboseLog::CriticalSection vlogLock;
 
       omrstr_ftime_ex(timestamp, sizeof(timestamp), "%b %d %H:%M:%S %Y", j9time_current_time_millis(), OMRSTR_FTIME_FLAG_LOCAL);
-      TR_VerboseLog::writeLineLocked(TR_Vlog_INFO, "StartTime: %s", timestamp);
+      TR_VerboseLog::writeLine(TR_Vlog_INFO, "StartTime: %s", timestamp);
+
+      TR_VerboseLog::write(TR_Vlog_INFO, "TimeZone: ");
+      if (-1 == omrstr_current_time_zone(&zoneSecondsEast, zoneName, sizeof(zoneName)))
```
Comment (keithc-ca) [inline runtime/compiler/runtime/Listener.cpp:359]: I think it's preferable to check for zero. In the future, the function may distinguish different failures with different non-zero return values. Same applies to the use in `javadump.cpp`.
```
if (0 != omrstr_current_time_zone(&zoneSecondsEast, zoneName, sizeof(zoneName)))
```

File: runtime/compiler/runtime/Listener.cpp [hunk 2]
```diff
@@ -348,9 +348,38 @@ static int32_t J9THREAD_PROC listenerThreadProc(void * entryarg)
...
+         TR_VerboseLog::write("(unavailable");
+      else
+         {
+         /* Writing UTC[+/-]hr:min (zoneName) to the log */
```
Comment (keithc-ca) [inline runtime/compiler/runtime/Listener.cpp:363]: This should say 'Write ...' and sentences end with a period.

File: runtime/compiler/runtime/Listener.cpp [hunk 3]
```diff
@@ -348,9 +348,38 @@ static int32_t J9THREAD_PROC listenerThreadProc(void * entryarg)
...
+         uint32_t offset = ((uint32_t)((zoneSecondsEast > 0) ? zoneSecondsEast : -zoneSecondsEast)) / 60;
+         uint32_t hours = offset / 60;
+         uint32_t minutes = offset % 60;
```
Comment (keithc-ca) [inline runtime/compiler/runtime/Listener.cpp:366]: These three declarations are not used (worse they're hidden by the equivalents below).

File: runtime/compiler/runtime/Listener.cpp [hunk 4]
```diff
@@ -348,9 +348,38 @@ static int32_t J9THREAD_PROC listenerThreadProc(void * entryarg)
...
+         TR_VerboseLog::write("UTC");
+
+         if (0 != zoneSecondsEast)
+            {
+            const char *format = (zoneSecondsEast > 0) ? "+%d" : "-%d";
+            uint32_t offset = ((uint32_t)((zoneSecondsEast > 0) ? zoneSecondsEast : -zoneSecondsEast)) / 60;
+            uint32_t hours = offset / 60;
+            uint32_t minutes = offset % 60;
```
Comment (keithc-ca) [inline runtime/compiler/runtime/Listener.cpp:374]: The type of these three variables should be `int32_t`; the coercion to unsigned is not necessary and does not match the formats being used.

File: runtime/compiler/runtime/Listener.cpp [hunk 5]
```diff
@@ -348,9 +348,38 @@ static int32_t J9THREAD_PROC listenerThreadProc(void * entryarg)
...
+
+            TR_VerboseLog::write(format, hours);
+            if (0 != minutes)
+               TR_VerboseLog::write(":%02d", minutes);
+            }
+         if ('\0' != *zoneName)
+            TR_VerboseLog::write(" (%s)", zoneName);
+         }
```
Comment (keithc-ca) [inline runtime/compiler/runtime/Listener.cpp:382]: I think a newline needs to be written to the log here.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: runtime/compiler/runtime/Listener.cpp (hunk 1 above)
Comment (mpirvu) [reply runtime/compiler/runtime/Listener.cpp:359]: JIT coding convention requires the `if` part to be enclosed in brackets, if the `else` part has brackets.

**Review by mpirvu — COMMENTED** [review]
I don't have any suggestions apart from the ones @keithc-ca already raised.

File: runtime/compiler/runtime/Listener.cpp (hunk 4 above)
Comment (cjjdespres) [reply runtime/compiler/runtime/Listener.cpp:374]: I took that from the existing code in `javadump.cpp`

https://github.com/eclipse-openj9/openj9/blob/master/runtime/rasdump/javadump.cpp#L704-L706

Would you like me to change them there too?

Comment (keithc-ca) [reply runtime/compiler/runtime/Listener.cpp:374]: No, the difference there is that the values will be passed to `writeInteger()` which has an unsigned parameter.

File: runtime/compiler/runtime/Listener.cpp (hunk 3 above)
Comment (cjjdespres) [reply runtime/compiler/runtime/Listener.cpp]: Sorry, forgot to remove those

**Comment by keithc-ca** [comment]
I expect the definition of `_LARGE_TIME_API` on line 25 of `javadump.cpp` can now be removed.

**Comment by cjjdespres** [comment]
It looks that way

**Comment by keithc-ca** [comment]
> I expect the definition of `_LARGE_TIME_API` on line 25 of `javadump.cpp` can now be removed.

And the include of `<time.h>` should no longer be needed.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
jenkins compile aix,osx,plinux,win,win32 jdk8

**Comment by keithc-ca** [comment]
Doh! Launching builds was premature: this depends on eclipse/omr#6309 which has not yet merged (let alone been promoted to the openj9 branch).

**Comment by keithc-ca** [comment]
jenkins compile aix,osx,plinux,win,win32 jdk8

**Pull request merged**